tests/unit/samplegen/test_samplegen.py

```diff
@@ -1048,6 +1317,175 @@ def test_validate_expression_base_attr_is_repeated():
     v.validate_expression("molluscs[0].mantle")
 
 
+def test_validate_expression_map_lookup():
+    # See https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto#L475
+    # for details on how mapped attributes get transformed by the protoc compiler.
+    OutputType = DummyMessage(
+        fields={
+            "cephalopods": DummyField(
+                message=DummyMessage(
+                    fields={
+                        # real type is most likely str in real protos
+                        "key": DummyField(),
+                        "value": DummyField(
+                            message=DummyMessage(
+                                fields={
+                                    "mantle": DummyField(
+                                        message=DummyMessage(type="MANTLE_TYPE",
+                                                             fields={}),
+                                    )
+                                },
+                                type="CEPHALOPOD_TYPE"
+                            )
+                        ),
+                    },
+                    type="CEPHALOPODS_TYPE",
+                    options=namedtuple("MessageOptions", ["map_field"])(True)),
+                repeated=True,
+            )
+        },
+        type="MOLLUSC_TYPE"
+    )
+    method = DummyMethod(output=OutputType)
+    v = samplegen.Validator(method)
+    v.validate_expression('$resp.cephalopods{"squid"}.mantle')
+
+
+def test_validate_expression_map_lookup_terminal_lookup():
+    OutputType = DummyMessage(
+        fields={
+            "cephalopods": DummyField(
+                message=DummyMessage(
+                    fields={
+                        "key": DummyField(),
+                        "value": DummyField(
+                            message=DummyMessage(
+                                fields={
+                                    "mantle": DummyField(
+                                        message=DummyMessage(type="MANTLE_TYPE",
+                                                             fields={}),
+                                    )
+                                },
+                                type="CEPHALOPOD_TYPE"
+                            )
+                        ),
+                    },
+                    type="CEPHALOPODS_TYPE",
+                    options=namedtuple("MessageOptions", ["map_field"])(True)),
+                repeated=True,
+            )
+        },
+        type="MOLLUSC_TYPE"
+    )
+    method = DummyMethod(output=OutputType)
+    v = samplegen.Validator(method)
+    v.validate_expression('$resp.cephalopods{"squid"}')
+
+
+def test_validate_expression_mapped_no_message():
```

Comment:
is this `no_message` or `non_map`? This looks identical to the next test case.....

Since it's hard to eyeball these testcases, could you, in general, add a comment to the lines that is expected to make the call fail? (eg the `map_field` set to `False`)

Reply:
Good catch. This one is an unintended, name-shadowing duplicate.

Reply:
Adding comments to the tests added in this PR. I'll annotate the others in a subsequent PR.